yt/fields/derived_field.py

```diff
@@ -34,6 +34,20 @@ def NullFunc(field, data):
     raise YTFieldNotFound(field.name)
 
 
+def DeprecatedFunc(ret_field, func):
+    def _DeprecatedFunc(field, data):
+        if data.ds.fields_detected:
+            args = [field.name]
+            msg = "The Derived Field %s is deprecated. "
+            if ret_field != field.name:
+                msg += "Use %s instead."
+                args.append(ret_field)
+            mylog.warning(msg, *args)
+        return func(field, data)
+
+    return _DeprecatedFunc
```

Comment:
I think this should use `functools.wraps` to support better tracebacks in case deprecated functions raise errors.
(the same change should be applied to TranslationFunc)

Another important point IMO is that a deprecation message should contain data about _when_ the deprecation was introduced and, more importantly, when it's due for removal.
This is already implemented as part of `yt._maintenance.deprecation.issue_deprecation_warning`, and it should probably be used here to avoid code duplication.

Reply:
@matthewturk what do you think about using `functools.wraps` here?

Reply:
@neutrinoceros let's make an issue for using `functools.wraps` here and do a separate PR.

Reply:
Is this snippet still relevant ? I remember seeing a different version where my second comment was taken into account already.

Reply:
The second comment, yes--but not the `functools.wraps` bit. I wanted to think a bit more about how to do the latter part, which may require a second PR. There may be warrant for a more general refactor here.

Reply:
fine by me :)